### PR TITLE
TICKET 0023894: Manage User Page - Search Section ALSO on PAGE TOP

### PR DESCRIPTION
**Pull request by fmancardi**
In addition, a help icon has been added to remember user using this feature that users can be searched by providing username or email.

### CODE REVIEW

File: manage_user_page.php [hunk 1]
```diff
@@ -362,7 +369,7 @@
 	<div id="manage-user-edit-div" class="form-inline pull-left">
 		<form id="manage-user-edit-form" method="get" action="manage_user_edit_page.php" class="form-inline"
 			<?php # CSRF protection not required here - form does not result in modifications ?>>
-			<label class="inline" for="username"><?php echo lang_get( 'search' ) ?></label>
+			<label class="inline" for="username"><?php echo lang_get( 'search' )?> <i class="fa fa-info-circle" title="<?php echo lang_get('user_search_hint')?>"></i> </label>
```
Comment (atrol) [inline manage_user_page.php]: I prefer placeholder as is works better with mobile devices, e.g. we use it for username on login page https://github.com/mantisbt/mantisbt/blob/master-2.10/login_page.php#L239

File: manage_user_page.php [hunk 2]
```diff
@@ -379,3 +386,33 @@
 </div>
 <?php
 layout_page_end();
+
+
+/**
+ *
+ */
```
Comment (atrol) [inline manage_user_page.php]: PHPDoc missing

File: manage_user_page.php (hunk 1 above)
Comment (fmancardi) [reply manage_user_page.php]: OK, got. Will remember it. Do you need I change this as do a new Pull Request? please let me know

File: manage_user_page.php [hunk 3]
```diff
@@ -379,3 +386,33 @@
...
+function draw_search_section($p_user_count,$p_per_page,$p_page_count, $p_page_number, $p_filter_string) {
```
Comment (atrol) [inline manage_user_page.php]: You introduce a new function that is quite similar to existing code.
The goal should be to replace existing code by a call to the new function.

Comment (fmancardi) [reply manage_user_page.php]: I've copied the existent code to create the function, OK if existent code can be replaced => two calls to this function. Please let me know how I've to proceed in order to provide work as will be better to be integrated on stable code base

**Comment by atrol** [comment]
Thanks for your contribution.

Please consider our Coding Guidelines, e.g. `One space after the opening ( and before the closing ) `
https://www.mantisbt.org/wiki/doku.php/mantisbt:coding_guidelines

File: manage_user_page.php (hunk 1 above)
Comment (atrol) [reply manage_user_page.php]: > Do you need I change this as do a new Pull Request?

No, you can go on committing/pushing to this one.
As a final step, we can squash all your commits to a single one.

File: manage_user_page.php [hunk 4]
```diff
@@ -358,24 +365,51 @@
...
+ * @param int    $p_page_count how many pages do we have.
+ * @param int    $p_page_number current page number.
+ * @param string $p_filter_string used by print_page_links() 
+ *
+ * @return void
+ */
+function draw_search_section($p_where,$p_user_count,$p_per_page,$p_page_count, $p_page_number, $p_filter_string) {
+
+	if( $p_user_count < $p_per_page ) {
+		return;
```
Comment (atrol) [inline manage_user_page.php]: Before this change, we were always able to search for a user by name.
Now there is use case where I am missing this functionality.

Assume there are 60 user names starting with character 'A'.
Click on `A` right beside the `All` button.
The search field is available in this case and you will be able to search for any user, not just the ones starting with 'A'.

Assume there are 40 user names starting with character 'A'.
Click on `A` right beside the `All` button.
The search field is not available in this case.

**Comment by fmancardi** [comment]
my fault.Will fix. check has to be done Only to hide search on TOP never on
Bottom.
Apologize

Il 29/Gen/2018 21:39, "Roland Becker" <notifications@github.com> ha scritto:

**Comment by atrol** [comment]
@fmancardi there were some enhancements in Manage User UI starting in version 2.17.0.
I think that having the search section on two places is no longer needed.

Do you agree that we can close this PR?

**Comment by atrol** [comment]
Closed as there was no more feedback.
@fmancardi feel free to reopen/rebase/resolve merge conflicts if you still see a need for the change after the enhancements starting in version 2.17.0.

**Pull request closed without merging**